src/appservice/Appservice.ts

```diff
@@ -470,6 +470,21 @@ export class Appservice extends EventEmitter {
         }
     }
 
+    /**
+     * Sets the visibility of a room in the appservice directory.
+     * @param {string} networkId The network ID to group the room under
+     * @param {string} roomId The room ID to manipulate the visibility of
+     * @param {"public" | "private"} visibility The visibility to set for the room
+     * @return {Promise} resolves when the visibility has been updated
+     */
+    public setAppserviceDirectoryVisibility(networkId: string, roomId: string, visibility: "public"|"private") {
```

Comment:
The name of this is duplicated:
```suggestion
    public setRoomDirectoryVisibility(networkId: string, roomId: string, visibility: "public"|"private") {
```

Reply:
This was intentional to differentiate from regular room directory shenanigans, but I guess since it's under `Appservice`